addons/html_editor/static/src/core/selection_plugin.js

```diff
@@ -432,15 +433,38 @@ export class SelectionPlugin extends Plugin {
         [focusNode, focusOffset] = normalizeFakeBR(focusNode, focusOffset);
         const selection = this.document.getSelection();
         if (selection) {
-            if (
-                selection.anchorNode !== anchorNode ||
-                selection.focusNode !== focusNode ||
-                selection.anchorOffset !== anchorOffset ||
-                selection.focusOffset !== focusOffset
-            ) {
-                selection.setBaseAndExtent(anchorNode, anchorOffset, focusNode, focusOffset);
+            if (this.canApplySelectionToDocument) {
+                if (
+                    selection.anchorNode !== anchorNode ||
+                    selection.focusNode !== focusNode ||
+                    selection.anchorOffset !== anchorOffset ||
+                    selection.focusOffset !== focusOffset
+                ) {
+                    selection.setBaseAndExtent(anchorNode, anchorOffset, focusNode, focusOffset);
+                }
+                this.activeSelection = this.makeActiveSelection(selection, true);
+            } else {
+                let range = new Range();
+                range.setStart(anchorNode, anchorOffset);
+                range.setEnd(focusNode, focusOffset);
+                if (anchorNode !== focusNode || anchorOffset !== focusOffset) {
+                    // Check if the direction is correct
+                    if (range.collapsed) {
+                        range = new Range();
+                        range.setEnd(anchorNode, anchorOffset);
+                        range.setStart(focusNode, focusOffset);
+                    }
+                }
```

Comment:
You can resolve whether the startContainer should be the anchor or the focus node by comparing their order in the DOM, instead of this trial and error approach: https://developer.mozilla.org/en-US/docs/Web/API/Node/compareDocumentPosition